tsl/src/hypercore/utils.c

```diff
@@ -31,13 +33,16 @@ hypercore_set_am(const RangeVar *rv)
 	HeapTuple tp;
 	Oid relid = RangeVarGetRelid(rv, NoLock, false);
 
-	tp = SearchSysCache1(RELOID, ObjectIdGetDatum(relid));
+	tp = SearchSysCacheLocked1(RELOID, ObjectIdGetDatum(relid));
```

Comment:
Shouldn't you use `SearchSysCacheLockedCopy1` since you're modifying the tuple?

Reply:
I think that is mostly to prevent "tuple already update" errors, which I don't think is an issue here. But it is probably a good idea to do it anyway just to be safe.

Reply:
Just mention it because the similar PostgreSQL code I find does this.